### PR TITLE
feat: add streetname approve endpoint

### PR DESCRIPTION
**Pull request by pgallik**

### CODE REVIEW

File: src/Common/Infrastructure/Controllers/ApiController.cs [hunk 1]
```diff
@@ -156,7 +156,11 @@ public abstract class ApiController<T> : PublicApiController
                     response.HeadersToKeyValuePairs(),
                     response.StatusCode);
             }
-            if (response.StatusCode is HttpStatusCode.OK or HttpStatusCode.Created or HttpStatusCode.Accepted)
+            if (response.StatusCode is HttpStatusCode.OK
+                or HttpStatusCode.Created
+                or HttpStatusCode.Accepted
+                or HttpStatusCode.PreconditionFailed
```
Comment (pgallik) [inline src/Common/Infrastructure/Controllers/ApiController.cs:162]: I had to add PreconditionFailed & NoContent to be able to get a correct response.

File: src/Public.Api/Public.Api.csproj [hunk 2]
```diff
@@ -44,6 +45,8 @@
     <Content Include="Dockerfile" CopyToOutputDirectory="Always" />
     <Content Include="appsettings.json" CopyToOutputDirectory="Always" />
     <Content Include="appsettings.*.json" CopyToOutputDirectory="Always" />
+    <None Remove="appsettings - Copy.json" />
```
Comment (lvermeulen) [inline src/Public.Api/Public.Api.csproj:48]: remove these 2 lines

File: src/Common/Infrastructure/Controllers/ApiController.cs (hunk 1 above)
Comment (ArneD) [reply src/Common/Infrastructure/Controllers/ApiController.cs:162]: `PreconditionFailed` is not an OK error code.
Checkout `HandleBadRequest` in the `RegistryApiController`.

Comment (pgallik) [reply src/Common/Infrastructure/Controllers/ApiController.cs:162]: This piece of code simply returns what it has received, HandleBadRequest throws an ApiException which has a completely different problem detail, is this the behaviour we expect from public api?

File: src/Public.Api/Public.Api.csproj (hunk 2 above)
Comment (lvermeulen) [reply src/Public.Api/Public.Api.csproj:48]: 2 lines are still there -- remove appsettings - Copy.json and appsettings.laptop-g7fahqcf.json

**Pull request merged**